lib/profiling/drhook/drhook.f90

```diff
@@ -32,21 +32,24 @@
 ! Authors J. Henrichs, Bureau of Meteorology
 
 
-!> An implemention of profile_mod which wraps the use of Dr Hook.
+!> An implemention of the PSyData API for profiling which wraps the use of Dr Hook.
```

Comment:
Yes, of course I did ... hang on ... oops. Well, I did test originally, but not after all my latest changes. And it is indeed broken. I've fixed the bugs here. I also have a new pull request more or less ready, that improves the profile example (gocean/eg5) to be able to use template, simple_timing, dl_timer, and drhook - i.e. it can create 4 binaries, that can all be executed. Since this is not really related to this, has a bit of documentation update, I moved this into a separate pull request to avoid adding more and more code/docs that need to be reviewed for this PR (see https://github.com/stfc/PSyclone/tree/improved_profiling_example, and #694).

Note that I cannot test the nvidia wrapper. I tried to install the library, but my VM is running out of disk space at this stage. It would be good if you could test this!

Reply:
I've fully compiled eg5 against the nvidia wrapper and it runs OK. However, it turns out that it does need to be running on a GPU so the profiling refuses to do anything. We could add profiling as an option to eg2 (since that does OpenACC) but that's something for #694 I think.

Reply:
Actually, eg2 does already have such an option but the code will not compile so I've created #697